packages/web-components/src/components/cta/video-cta-composite.ts

```diff
@@ -90,9 +90,12 @@ class DDSVideoCTAComposite extends ModalRenderMixin(
   // @ts-ignore: The decorator refers to this method but TS thinks this method is not referred to
   private async _handleRequestVideoData(event: CustomEvent) {
     const { href, videoName: customVideoName, videoDescription } = event.detail;
+    // Use a default value for thumbnail width in case the element doesn't have
+    // an offsetWidth yet.
+    const thumbnailWidth = (event?.target as HTMLElement)?.offsetWidth || 400;
```

Comment:
i wonder if we should use the same logic here: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/7914... set the default to 3 so it's a blurry thumbnail on load?

Reply:
Hum, maybe. I need to study how it works to eventually replace the default 3px width with the one that's needed. Here it's set once when the components mount and sometimes that so quick the parent element doesn't have a defined width yet. I'll look into it tomorrow and get back. Thanks.

Reply:
Ok, so couple of things here. I've gone with a slightly different approach, but really just changed where the same thing I was doing here is happening.

* I've moved the determination of the `videoThumbnailUrl` to the more natural place IMO, the `VideoCTAMixin`. The `videoThumbnailUrl` doesn't depend on the async Kaltura API request that's made here. It's put together sync using the local `KalturaPlayerAPI.getThumbnailUrl` method. That also allows us to change it according to the component width when the component updates. This [matches `video-player`](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/packages/web-components/src/components/video-player/video-player.ts#L276-L278) in terms of when the video thumbnail url is set and should avoid the issue that we are seeing.
* The logic in https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/7914 is different than here. There the `<dds-video-player>` is managed by us and is initiated with a `thumbnailUrl` attribute, and [when it's put together](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/packages/web-components/src/components/video-player/video-player-composite.ts#L295), the component isn't mounted yet, so we don't know what size it will need, so we start with a tiny width of 3px. It then [updates it's thumbnail on `updated`](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/packages/web-components/src/components/video-player/video-player.ts#L276-L278) when the component has rendered and has an positive `offsetWidth`.
* Here, we're not the ones creating the `<dds-card-group-item>`, that's the user, and it's up to them to initiate with a thumbnail (or not). Whether we get a `video-thumbnail-url` attribute or not, our code sets it using the Kaltura API (this _appears_ to be by design). The user provides a custom thumbnail using the `thumbnail` attribute, which is used first if present. Since we set the video thumbnail url, it's done in `updated`, when we know the `offsetWidth` and can straight away ask for the right size once from Kaltura.

I hope that makes sense. Let me know if anything is unclear.